hs628_tools/feature_missing.py

```diff
@@ -0,0 +1,201 @@
+                            Train data, observations and features
+
+                target:     numpy ndarrays, shape = (1, ) target can be binary if target_type is classification or
+                            continuous if target_type is regression
+
+                target_type : string defines the RandomForest method if target is continuous or binary. default value
+                            = "classification"
+
+                importance_thresh : float between 0 and 1
+                            Pre-defined threshold set by the user, default value = 1/number of the features
+            -----------
+            Returns:
+                A ranked (sorted) list containing indices of features  with importance value below importance_thresh
+
+        """
+        # Selecting the model based on target_type
+
+        if target_type == "regression":
+            model = RandomForestRegressor()
+        elif target_type == "classification":
+            model = RandomForestClassifier()
+        else:
+            raise ValueError('Target type must be "regression" or "classification" ')
+
+        feature_importance = [0] * features.shape[1]
+        # creating k-fold cross validation with 10 folds
+        skf = StratifiedKFold(n_splits=10, random_state=None, shuffle=False)
+        for train_index, test_index in skf.split(features, target):
+            features_train = features[train_index]
+            targets_train = target[train_index]
+            features_test = features[test_index]
+            targets_test = target[test_index]
+            # Fitting the model
+            model.fit(features_train, targets_train)
+            model.score(features_test, targets_test)
+            feature_importance += model.feature_importances_
+        # normalizing the feature importance values
+        feat_imp_norm = feature_importance / skf.get_n_splits(features, target)
+        # Creating a template (copy) to sort low important features
+        feat_imp_temp = feat_imp_norm.copy()
+        low_imp = []
+
+        for i in range(feat_imp_temp.shape[0]):
+            # finding the index of the minimum value and save it to low_importance
+            low_imp += [feat_imp_temp.argmin()]
+            # set the minimum to a large number in feat_imp_norm to find the next minimum value
+            feat_imp_temp[feat_imp_temp.argmin()] = 1000
+
+        low_importance = []
+        for i in range(len(low_imp)):
+            if feat_imp_norm[low_imp[i]] < importance_thresh:
+                print('Column #%d is #%d in the list of low importance features with the importnace value of %s \n' % (
+                    low_imp[i], i + 1, feat_imp_norm[low_imp[i]]))
+                low_importance += [low_imp[i]]
+
+        self.low_importance_col = np.array(low_importance)
+        return self.low_importance_col
+
+
+if __name__ == "__main__":
+    data_np = np.genfromtxt('data.csv', delimiter=',', skip_header=1, missing_values='None', filling_values=np.nan)
```

Comment:
Hi Omid, I definitely do not have any comments regarding code quality. I think you done a great job. Let me nitpick few things which you could improve open : I noticed you had multiple for loops embedded in the code for correlation, I wonder if there is any way to avoid that as you see it basically adds a lot of run-time complexity to your code when you have a relatively big dataset. You have two inner for loops which to think there is a way to avoid repetition using zip in python, I'm not entirely sure if it's feasible but you can try. Also you are loading the hard-coded name of the dataset which I earlier did too, so a suggestion to load the filepath name through command line argument like I did to just avoid any hard-coding of parameters which you have done efficiently throughout the rest of your code.